admin/controllers/authentication.js

```diff
@@ -27,65 +54,107 @@ const getSignIn = (req, res) => {
   if (req.isAuthenticated()) {
     res.redirect('/school/school-home')
   } else {
-    if (config.NCA_TOOLS_AUTH_URL) {
-      res.redirect(config.NCA_TOOLS_AUTH_URL)
-    } else {
-      res.render('sign-in')
-    }
+    redirectToAuthModeSignIn(res)
   }
 }
 
-const postSignIn = (req, res) => {
-  // Only id is available from local and NCA auth
-  const { displayName, id, role, timezone } = req.user
-  logger.info(`postSignIn: User ID logged in: ${id} (${displayName}) timezone is "${timezone}"`)
+const postDfeSignIn = async (req, res) => {
+  req.user = await dfeSignInService.process(req.user)
+  logger.info(`postSignIn: User ID logged in:
+    id:${req.user.id} \n
+    displayName:${req.user.displayName} \n
+    role:${req.user.role} \n
+    timezone:"${req.user.timezone}"`)
 
-  switch (role) {
+  switch (req.user.role) {
     case 'TEACHER':
     case 'HEADTEACHER':
+    case 'HELPDESK':
       return res.redirect(homeRoutes.schoolHomeRoute)
     case 'TEST-DEVELOPER':
       return res.redirect(homeRoutes.testDeveloperHomeRoute)
     case 'SERVICE-MANAGER':
       return res.redirect(homeRoutes.serviceManagerHomeRoute)
+    default:
+      return res.redirect(homeRoutes.schoolHomeRoute)
+  }
+}
+
+const postSignIn = (req, res) => {
+  logger.info(`postSignIn: User ID logged in:
+    id:${req.user.id} \n
+    displayName:${req.user.displayName} \n
+    role:${req.user.role} \n
+    timezone:"${req.user.timezone}"`)
+
+  switch (req.user.role) {
+    case 'TEACHER':
+    case 'HEADTEACHER':
     case 'HELPDESK':
```

Comment:
Don't the helpdesk have their own page now?

Reply:
indeed they do.  I tested the helpdesk role yesterday and it worked, however I will walk through the code today to ensure its valid.  Maybe it does a redirect after establishing that no school impersonation is set...

Reply:
the implementation in `master` from @GeorgeChatzigiannis recent PR always routes helpdesk to the school home page, which in turn redirects to the impersonation page, if necessary - https://github.com/DFEAGILEDEVOPS/MTC/blob/43b8288ceb7ba4c374a59140317669c60e4b8f4d/admin/controllers/school.js#L27

Seems like a valid solution to me